lib/Arch/SPARC32_C.cpp

```diff
@@ -0,0 +1,347 @@
+        return llvm::Error::success();
+
+      // Split the integer across `o5:o0`. Experimentally, the largest
+      // returnable integer is 192 bits in size, any larger and LLVM crashes.
+      } else if (bit_width <= 192) {
+        const char *ret_names[] = {"o0", "o1", "o2", "o3", "o4", "o5"};
+        for (auto i = 0u; i < 6 && (32 * i) < bit_width; ++i) {
+          ret_values.emplace_back();
+          auto &value_declaration = ret_values.back();
+          value_declaration.reg = arch->RegisterByName(ret_names[i]);
+          value_declaration.type = int32_ty;
+        }
+        return llvm::Error::success();
+      } else {
+        return llvm::createStringError(
+            std::errc::invalid_argument,
+            "Could not allocate integral type '%s' to return register(s) in function '%s'",
+            remill::LLVMThingToString(ret_type).c_str(),
+            function.getName().str().c_str());
+      }
+    }
+
+    // Pointers always fit into `EAX`.
+    case llvm::Type::PointerTyID: {
+      ret_values.emplace_back();
+      auto &value_declaration = ret_values.back();
+      value_declaration.reg = arch->RegisterByName("o0");
+      value_declaration.type = ret_type;
+      return llvm::Error::success();
+    }
+
+    case llvm::Type::HalfTyID:
+    case llvm::Type::FloatTyID: {
+      ret_values.emplace_back();
+      auto &value_declaration = ret_values.back();
+      value_declaration.reg = arch->RegisterByName("f0");
+      value_declaration.type = ret_type;
+      return llvm::Error::success();
+    }
+
+    case llvm::Type::DoubleTyID: {
+      ret_values.emplace_back();
+      auto &value_declaration = ret_values.back();
+      value_declaration.reg = arch->RegisterByName("d0");
+      value_declaration.type = ret_type;
+      return llvm::Error::success();
+    }
+
+    // Try to split the composite type over registers, and fall back on RVO
+    // if it's not possible.
+    case llvm::GetFixedVectorTypeId():
+    case llvm::Type::ArrayTyID:
+    case llvm::Type::StructTyID: {
+      AllocationState alloc_ret(return_register_constraints, arch, this);
+      alloc_ret.config.type_splitter = IntegerTypeSplitter;
+      auto mapping = alloc_ret.TryRegisterAllocate(*ret_type);
+
+      // There is a valid split over registers, so add the mapping
+      if (mapping) {
+        return alloc_ret.CoalescePacking(mapping.getValue(), ret_values);
+
+      // Composite type splitting; Unlike with x86, LLVM doesn't naturally
+      // perform RVO on large structures returned by value from bitcode.
+      } else {
+        break;
+      }
+    }
+
+    default:
+      break;
+  }
+
+  return llvm::createStringError(
+      std::errc::invalid_argument,
+      "Could not allocate unsupported type '%s' to return register in function '%s'",
+      remill::LLVMThingToString(ret_type).c_str(),
+      function.getName().str().c_str());
+}
+
+// For X86_64_SysV, the general argument passing behavior is, try to pass the
```

Comment:
This comment needs to be updated to reflect SPARC things; its left over as copypasta from x86-64